### ISSUE TITLE
guess_readme() could return multiple README paths

### ISSUE THREAD

**Issue by bastistician** [ropensci/codemetar#190]
`.guess_readme()` currently does not ensure that the returned `readme_path` contains only one element. An example is #189. After fixing that issue there could still be something like files `README.md` and `DO_NOT_README.md` ;-), which would both be matched and returned in the `readme_path` vector. However, `give_opinions_readme()`, which is called from `give_opinions()` on the result of `guess_readme()`, assumes `readme_path` to be of length one. Here is the traceback of calling `give_opinions()` in a (hypothetical) package root containing two such "README" files:

```R
traceback()
## 7: file(con, "r")
## 6: readLines(path)
## 5: `_f`(path = path)
## 4: withVisible(`_f`(path = path))
## 3: extract_badges(readme_path)
## 2: give_opinions_readme(readme, desc_info$identifier)
## 1: give_opinions()
```

Two suggestions:

1. Adding a `^` prefix (and `$` suffix, #189) to the regular expression at
https://github.com/ropensci/codemetar/blob/387456e9d62fdb9be936a089d7173e8d21d783b8/R/guess_metadata.R#L172
will make it unlikely that two README files are returned (only possible on case-sensitive systems), but would no longer match README files with a prefix (I'm not sure if it should).

2. Either ensure that `.guess_readme()$readme_path` is always NULL or of length 1 (e.g., checking for final `length(readme_path) > 1` in which case only the first element is returned with a warning), or defer that step to all functions calling `guess_readme()`, if `guess_readme()` itself is designed to return all potential candidates of README files.

**Comment by maelle** [comment]
Thanks a ton!

>  but would no longer match README files with a prefix (I'm not sure if it should).

Yeah I'm not sure it should, unless you know of examples where it'd make sense?

**Comment by bastistician** [comment]
Not really. The base package [**stats**](https://github.com/wch/r-source/tree/trunk/src/library/stats) has `loess-README` and `splines-README`, but these are not in the intended markdown format anyway. I think I have seen software projects with files like README_subject1 and README_subject2, but not R packages. 

IMHO, `guess_readme()` is intended to find the single README file of an R package which contains the badges to be inspected by the other guess-functions. And I think it is reasonable to confine the (case-insensitive) search to files called `README.Rmd` or `README.md`.

**Comment by maelle** [comment]
Yes, you're right! Do you mind making a PR? If you do, please add yourself as "ctb" (with your ORCID ID! I'd recommend doing that via `desc::desc_add_me()` cf https://ropensci.org/technotes/2018/10/08/orcid/ )

**Comment by maelle** [comment]
but we need to account for `readme.rmd` and so hence the case-insentivity btw.
